sklearn/cluster/_optics.py

```diff
@@ -34,10 +34,10 @@ class OPTICS(ClusterMixin, BaseEstimator):
     """Estimate clustering structure from vector array.
 
     OPTICS (Ordering Points To Identify the Clustering Structure), closely
-    related to DBSCAN, finds core sample of high density and expands clusters
-    from them [1]_. Unlike DBSCAN, keeps cluster hierarchy for a variable
+    related to DBSCAN, finds core samples of high density and expands clusters
+    from them [1]_. Unlike DBSCAN, it keeps cluster hierarchy for a variable
     neighborhood radius. Better suited for usage on large datasets than the
-    current sklearn implementation of DBSCAN.
+    current scikit-learn implementation of DBSCAN.
 
     Clusters are then extracted using a DBSCAN-like method
     (cluster_method = 'dbscan') or an automatic
```

Comment:
In the paragraph below, does the sentence:

"then computing only the distances to unprocessed points when constructing the cluster order"

mean that the original OPTICS algorithm does not only compute unprocessed points, or is it saying that the KNN part differs and this is just the step that follows (but does not differ from the original OPTICS)?

Reply:
the original OPTICS algorithm proposed in the paper calculates core distance and reachability distances for all objects as a first step, which we don't. It's not easy to explain really, I had a hard time converting that to code.

Reply:
I guess this comment is not for the average user, and rather for somebody picky who reads the paper and then comes looks at the implementation here. You could argue it could be a comment in the code maybe.

Reply:
Ahh I think I understand the meaning now.

Does "unprocessed point" mean the same in both implementations? Does the original OPTICS algorithm re-compute reachability distances, updating if they are smaller? i.e., is the second part of the algorithm the same here as in the original paper?

Reply:
Oh I'd have to read the paper and compare with implementation to be able to answer that, can't really confidently respond to this question easily

Reply:
Yeah fair, I just looked at it and am still confused :upside_down_face: